unittests/allplatformstests.py

```diff
@@ -4222,6 +4222,20 @@ def do_install(tags, expected_files, expected_scripts):
         do_install('runtime,custom', expected_runtime_custom, 1)
         do_install(None, expected_all, 2)
 
+
+    def test_install_script_dry_run(self):
+        testdir = os.path.join(self.common_test_dir, '53 install script')
+        self.init(testdir)
+        self.build()
+
+        cmd = self.meson_command + ['install', '--dry-run', '--destdir', self.installdir]
+        self._run(cmd, workdir=self.builddir)
+
+        installpath = Path(self.installdir)
+        self.assertTrue((installpath / 'usr/diiba/daaba/file.dat').exists())
```

Comment:
nitpicking: This is testing what *not* to do: creating files on dry run. I think the script should be modified to check for MESON_INSTALL_DRY_RUN env (which is untested atm) and instead print something on stdout that can be verified here.